### DISCUSSION TITLE
Uploading or embed pdfs

### DISCUSSION THREAD

**Discussion by mrchrisadams** [outline/outline#1788] [Ideas]
Hi there,

I had an assumption that if I could upload images to outline, I'd also be able to upload PDFs, but this turns out not to be the case.

While I know a key idea for outline is having a searchable knowledge base, I found myself wishing I could upload PDFs pretty quickly, for work that we don't change, might want to be able to access in future.

You could even make it searchable, presumably - either with brute force, by searching using RGA each time across all the PDFs (probably not a good idea):

https://phiresky.github.io/blog/2019/rga--ripgrep-for-zip-targz-docx-odt-epub-jpg/

Or something more sensible, like pulling out the text on upload with PDF.js, and dumping that into the full-text search index provided by Postgres. 
https://www.garysieling.com/blog/building-a-full-text-index-in-javascript/

I had a quick look around, but I couldn't figure out where the model for sequelize was, to see what database features of postgres are in use.

**Comment by mrchrisadams** [comment]
Oh hang on, I've just found this issue here. It seems to cover it.

https://github.com/outline/outline/issues/859

Feel free to close 👍

**Reply by tommoor** [reply]
Hey, yea I'd say that issue covers the upload portion but not making the contents of those files searchable – which hasn't been thought about too much yet. It does seem like you could extract the content and store it against the `attachment` model in a full text index like you suggest.

I'm interested in what kind of content would be in the PDF's and would you be happy in your usecase to see them treated as file attachments to a document? We don't want to slip into becoming a file host like Dropbox or GDrive here…

**Reply by mrchrisadams** [reply]
TBH, it was more a case of the text appearing in outline, when hitting CMD+K then searching, because uploading it made the content visible to that index.

I really like the UX of the outline searching in that fashion, and if there's a PDF document that isn't easy to reformat because the layout is a key part, at being able to at least find that PDF based on the text within is useful.

**Reply by mrchrisadams** [reply]
To provide even more context, this might be finding content from a bunch reports quickly, that a group of researchers might sifting through.

**Comment by matbrgz** [comment]
This could work to, embed use google drive. https://github.com/outline/outline/pull/1702

**Comment by aam1r** [comment]
I added some more use cases (other than PDFs) at https://github.com/outline/outline/discussions/1849. 

As a nice MVP, I am OK with it not being embedded. It can just show a hyperlink to the uploaded files.
